compiler/src/iree/compiler/Codegen/Common/EmulateNarrowType.cpp

```diff
@@ -95,6 +95,202 @@ static void populateIreeNarrowTypeEmulationPatterns(
                                                   patterns.getContext());
 }
 
+static bool isByteAligned(ShapedType type) {
+  unsigned elementBits = type.getElementType().getIntOrFloatBitWidth();
+  auto numElements = type.getNumElements();
+  return (numElements * elementBits) % 8 == 0;
+}
+
+struct PadSubbyteTransferWritePattern
```

Comment:
I dont know if we need these patterns. What are they for?

Reply:
turns out those are not needed. removed.